mlflow/models/evaluation/validation.py

```diff
@@ -0,0 +1,202 @@
+    def min_relative_change(self):
+        """
+        Minimum relative change required to pass model comparison wwith baseline model
+        :rtype: float
+        """
+        return self._min_relative_change
+
+    @property
+    def higher_is_better(self):
+        """
+        Whether higher value is better for the metric.
+        :rtype: bool
+        """
+        return self._higher_is_better
+
+    def is_empty(self):
+        """
+        Return True if there is no threshold specified, False otherwise.
+        """
+        return (
+            self._threshold is None
+            and self._min_absolute_change is None
+            and self._min_relative_change is None
+        )
+
+    def __str__(self):
+        """
+        Return a human-readable string consisting of all specified thresholds.
+        """
+        threshold_strs = []
+        if self._threshold is not None:
+            threshold_strs.append(f"Threshold: {self._threshold}.")
+        if self._min_absolute_change is not None:
+            threshold_strs.append(f"Minimum Absolute Change: {self._min_absolute_change}.")
+        if self._min_relative_change is not None:
+            threshold_strs.append(f"Minimum Relative Change: {self._min_relative_change}.")
+        if self._higher_is_better is not None:
+            if self._higher_is_better:
+                threshold_strs.append("Higher value is better.")
+            else:
+                threshold_strs.append("Lower value is better.")
+        return " ".join(threshold_strs)
+
+
+class _MetricValidationResult:
+    """
+    Internal class for representing validation result per metric.
+    Not user facing, used for organizing metric failures and generating failure message
+    more conveniently.
+    :param metric_name: String representing the metric name
+    :param metric_threshold: :py:class: `MetricThreshold<mlflow.models.validation.MetricThreshold>`
+                             The MetricThreshold for the metric.
+    """
+
+    missing_candidate = False
+    missing_baseline = False
+    threshold_failed = False
+    min_absolute_change_failed = False
+    min_relative_change_failed = False
+
+    def __init__(
+        self,
+        metric_name,
+        candidate_metric_value,
+        metric_threshold,
+        baseline_metric_value=None,
+    ):
+        self.metric_name = metric_name
+        self.candidate_metric_value = candidate_metric_value
+        self.baseline_metric_value = baseline_metric_value
+        self.metric_threshold = metric_threshold
+
+    def __str__(self):
+        """
+        Return a human-readable string representing the validation result for the metric.
+        """
+        if self.is_success():
+            return f"Metric {self.metric_name} passed the validation."
+
+        if self.missing_candidate:
+            return (
+                f"Metric {self.metric_name} was missing from the "
+                f"evaluation result of the candidate model."
```

Comment:
Can you change this to the format of `"<Failure>: <reason>"`? Including metric name in `<Failure>`.